### DISCUSSION TITLE
Randomly occurring: The app encountered an error processing your request.

### DISCUSSION THREAD

**Discussion by IainD925** [microsoft/AI-Chat-App-Hack#135] [Q&A] [answered]
`Error type: <class 'openai.BadRequestError'>`
Happening in both Chat & Ask.  I ask the same bunch of questions several times and randomly get this error on a previously asked question.  Refreshing doesn't fix but appears to resolve itself after a period of time.

Logs don't show anything of note.

Anyone else seeing this behaviour?

**Comment by IainD925** [comment, marked as answer]
Found the answer in Application Insights - Errors.

> This model's maximum context length is 4096 tokens. However, your messages resulted in 14821 tokens. Please reduce the length of the messages.

I increased the chunking size and when you ask for multiple sources, the response is additive.  Reduced to 1 source and issue resolved.

**Reply by pamelafox** [reply]
Interesting, we have logic that tries to truncate the old conversation to fit in the message history, so it should have just truncated to 4096. However, it will fail if even a single question's sources exceed the token window. I made a lil video showing that message truncation logic- https://youtu.be/GIuj_w2HlT0
Glad you figured out a workaround!
